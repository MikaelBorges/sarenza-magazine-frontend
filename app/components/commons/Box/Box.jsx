import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import styles from './Box.module.scss';

const Box = ({ children, smallMargin, displayColumn, hasError, placeholder }) => (
  <div
    className={classnames(styles.box, {
      [styles.smallMargin]: smallMargin,
      [styles.displayColumn]: displayColumn,
      [styles.error]: hasError
    })}>
    {children || <div className={styles.placeholder}>{placeholder}</div>}
  </div>
);

Box.propTypes = {
  /**
   * Component or Text wrapped on the box
   */
  children: PropTypes.node,
  /**
   * Has small margin
   */
  smallMargin: PropTypes.bool,
  /**
   * display column
   */
  displayColumn: PropTypes.bool,
  /**
   * if has error the background become red
   */
  hasError: PropTypes.bool,
  /**
   * Placeholder of the box if hasnt children
   */
  placeholder: PropTypes.string
};

Box.defaultProps = {
  children: '',
  smallMargin: false,
  displayColumn: false,
  hasError: false,
  placeholder: ''
};

export default Box;
