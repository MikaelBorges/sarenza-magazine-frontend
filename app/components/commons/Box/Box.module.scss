@import '../../_styles/theme';

$box-error-bg-color: pink !default;
$box-error-border-color: red !default;
$box-placeholder-color: #aaa !default;

.box {
  display: flex;
  margin-bottom: 20px;
  min-height: 40px;
  background: #fff;
  border: 1px solid #d5d5d5;
  border-radius: 10px;
  padding: 10px 20px;

  &.error {
    background-color: $box-error-bg-color;
  }
}
.smallMargin {
  margin-bottom: 10px;
}
.displayColumn {
  flex-direction: column;
}
.placeholder {
  font-style: italic;
  color: $box-placeholder-color;
}
