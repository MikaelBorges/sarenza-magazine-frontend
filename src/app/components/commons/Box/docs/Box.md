# Box

Simple rectangular container.

## When To Use

A box can be used to display content related to a single subject. The content can consist of multiple elements of varying types and sizes.

### Example

For import the component

```js
import Box from 'sarenza-uikit/Box';
```

<!-- SOURCE -->

### Result

<!-- STORY -->

### Props

<!-- PROPS -->
