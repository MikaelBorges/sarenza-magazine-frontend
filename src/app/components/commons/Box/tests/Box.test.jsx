import '@testing-library/jest-dom/extend-expect';

import { render } from '@testing-library/react';
import * as React from 'react';

import Box from '../Box';

describe('<Box />', () => {
  it('should match the expected snapshot with default props', () => {
    const { container, getByText } = render(<Box>test</Box>);

    expect(getByText('test')).toBeDefined();
    expect(container).toMatchSnapshot();
  });
  it('should match the expected snapshot without children but with placeholder', () => {
    const { container, getByText } = render(<Box placeholder="toto" />);
    expect(getByText('toto')).toBeDefined();
    expect(container).toMatchSnapshot();
  });
  it('should match the expected snapshot with complete css class', () => {
    const { container } = render(<Box smallMargin displayColumn hasError />);

    expect(container.querySelector('.smallMargin')).toBeDefined();
    expect(container.querySelector('.displayColumn')).toBeDefined();
    expect(container.querySelector('.error')).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
