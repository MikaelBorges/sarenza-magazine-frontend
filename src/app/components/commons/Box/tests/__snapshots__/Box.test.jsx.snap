// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> should match the expected snapshot with complete css class 1`] = `
<div>
  <div
    class="box smallMargin displayColumn error"
  >
    <div
      class="placeholder"
    />
  </div>
</div>
`;

exports[`<Box /> should match the expected snapshot with default props 1`] = `
<div>
  <div
    class="box"
  >
    test
  </div>
</div>
`;

exports[`<Box /> should match the expected snapshot without children but with placeholder 1`] = `
<div>
  <div
    class="box"
  >
    <div
      class="placeholder"
    >
      toto
    </div>
  </div>
</div>
`;
